### PR TITLE
Ttd Bid adapter: support for transaction id, bcat, and consolidate params

### PR DESCRIPTION
**Pull request by julian-burger-ttd**
## Type of change

- [X] Feature
- [X ] Does this change affect user-facing APIs or examples documented on http://prebid.org?

## Description of change
TTD Adapter Updates

Remove siteId parameter
Make placementId parameter optional if GPID is passed
Pass transactionId through to source.tid
Read bcat from ortb2

### CODE REVIEW

File: modules/ttdBidAdapter.js [hunk 1]
```diff
@@ -58,7 +58,9 @@ function getBidFloor(bid) {
 }
 
 function getSource(validBidRequests) {
-  let source = {};
+  let source = {
+    tid: validBidRequests[0].transactionId
```
Comment (patmmccann) [inline modules/ttdBidAdapter.js:62]: @JudSpencer mentioned in ortb slack he would prefer if you guys followed Rob's proposal and implementation here https://github.com/prebid/Prebid.js/pull/8641/files to gather imp.ext.tid

File: modules/ttdBidAdapter.js [hunk 2]
```diff
@@ -387,6 +382,10 @@ export const spec = {
       ext: getExt(firstPartyData)
     }
 
+    if (firstPartyData && firstPartyData.bcat) {
+      topLevel.bcat = firstPartyData.bcat;
```
Comment (patmmccann) [inline modules/ttdBidAdapter.js:386]: can you document if this overwrites or appends existing settings for an account on the backend? I think append is not particularly workable, and overwrite is preferred.

Why not battr and badomain as well?

File: modules/ttdBidAdapter.js (hunk 1 above)
Comment (patmmccann) [reply modules/ttdBidAdapter.js:62]: adding to this, source.tid as the first transaction id makes sense to me, but we also seem to have debate over whether it should be the auction id, and implementors have chosen one or the other. Prebid Server seems to just make up its own id. So aside from the cardinality issue that imp.ext.tid addresses, we have other issues with adapters defining it differently (we have open issues discussing). Imp.ext.tid shares none of these concerns, only adoption, hence me encouraging you to pick it up here.

File: modules/ttdBidAdapter.js (hunk 2 above)
Comment (patmmccann) [reply modules/ttdBidAdapter.js:386]: Sorry, badv

Comment (markd-fs) [reply modules/ttdBidAdapter.js:386]: We would also like to see support for battr and badv here.

Comment (julian-burger-ttd) [reply modules/ttdBidAdapter.js:386]: I'll add battr/badv. We're still discussing behavior internally.

Comment (patmmccann) [reply modules/ttdBidAdapter.js:386]: thanks!

Comment (julian-burger-ttd) [reply modules/ttdBidAdapter.js:386]: We decided to make the blocking additive (docs will reflect this).

Comment (patmmccann) [reply modules/ttdBidAdapter.js:386]: Can we discuss? This decision breaks all utility of this feature

Comment (patmmccann) [reply modules/ttdBidAdapter.js:386]: @julian-burger-ttd can you confirm the badv battr changes above are planned for this pr?

Comment (julian-burger-ttd) [reply modules/ttdBidAdapter.js:386]: Confirming for this PR. Haven't had a chance to get to it yet, though. As for overwrite vs. additive we having some internal discussion as there are multiple use cases to consider.

File: modules/ttdBidAdapter.js (hunk 1 above)
Comment (julian-burger-ttd) [reply modules/ttdBidAdapter.js]: I updated this but still new to the area so please verify it is what you expected.

File: modules/ttdBidAdapter.js (hunk 2 above)
Comment (julian-burger-ttd) [reply modules/ttdBidAdapter.js:397]: Updated the code. Docs updates (separate MR anyway) will follow.

File: modules/ttdBidAdapter.js [hunk 3]
```diff
@@ -58,7 +58,9 @@ function getBidFloor(bid) {
 }
 
 function getSource(validBidRequests) {
-  let source = {};
+  let source = {
+    tid: utils.deepAccess(validBidRequests[0], 'ortb2Imp.ext.tid')
```
Comment (patmmccann) [inline modules/ttdBidAdapter.js:62]: you had source.tid fine; but what I was imagining is that you would want something like 

https://github.com/prebid/Prebid.js/blob/d46661a4aa8994d57f3bea3055bd4e4c8aa50e10/modules/ixBidAdapter.js#L248 / https://github.com/prebid/Prebid.js/blob/d46661a4aa8994d57f3bea3055bd4e4c8aa50e10/modules/ixBidAdapter.js#L361

as you guys are asking other exchanges for that in other forums

File: modules/ttdBidAdapter.js [hunk 4]
```diff
@@ -141,17 +142,21 @@ function getSite(bidderRequest, firstPartyData) {
 
 function getImpression(bidRequest) {
   let impression = {
-    id: bidRequest.bidId,
-    tagid: bidRequest.params.placementId
+    id: bidRequest.bidId
   };
 
-  let gpid = utils.deepAccess(bidRequest, 'ortb2Imp.ext.gpid');
+  const gpid = utils.deepAccess(bidRequest, 'ortb2Imp.ext.gpid');
```
Comment (patmmccann) [inline modules/ttdBidAdapter.js:148]: right here is where you could also get ext.tid

File: modules/ttdBidAdapter.js (hunk 3 above)
Comment (patmmccann) [reply modules/ttdBidAdapter.js:62]: I commented below on where it would go

**Review by patmmccann — CHANGES_REQUESTED** [review]
comments on imp.ext.tid vs source.tid

File: modules/ttdBidAdapter.js (hunk 3 above)
Comment (julian-burger-ttd) [reply modules/ttdBidAdapter.js]: Thanks I think I understand what you are getting at now.

**Comment by patmmccann** [comment]
Lgtm

**Review by patmmccann — APPROVED** [review]

**Pull request merged**